runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -23,9 +23,16 @@
 #if !defined(BYTECODEINTERPRETER_HPP_)
 #define BYTECODEINTERPRETER_HPP_
 
-#if JAVA_SPEC_VERSION >= 20
+#if JAVA_SPEC_VERSION >= 21
 #include <errno.h>
-#endif /* JAVA_SPEC_VERSION >= 20 */
+#if defined(WIN32)
+/* Ignore the definition of UDATA as it is defined in <windows.h>. */
+#define UDATA UDATA_win32_
+#include <windows.h>
+#undef UDATA /* This is safe because our UDATA is a typedef rather than a macro. */
+#include <winsock2.h>
+#endif /* WIN32 */
```

Comment:
The comment should repeat the `#if` condition:
```c
#endif /* defined(WIN32) */
```
Same comment for lines 5296 and 5298.

Reply:
Fixed.